### PR TITLE
Fixed UI according to supporting dynamic font type

### PR DESCRIPTION
**Pull request by EgorovEI**

### CODE REVIEW

**Review by dukhovnyi — APPROVED** [review]

File: GliaWidgets/ViewController/Chat/ChatViewController.swift [hunk 1]
```diff
@@ -23,6 +23,12 @@ class ChatViewController: EngagementViewController, MediaUpgradePresenter,
         viewModel.event(.viewDidLoad)
     }
 
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+
+        (self.view as? ChatView)?.messageEntryView.updateLayout()
```
Comment (igorkravchenko) [inline GliaWidgets/ViewController/Chat/ChatViewController.swift:29]: Let's add comment here why this has to be done.

File: GliaWidgets/View/Chat/Entry/Upload/FileUploadListView.swift [hunk 2]
```diff
@@ -10,7 +11,13 @@ class FileUploadListView: UIView {
     private let stackView = UIStackView()
     private var heightLayoutConstraint: NSLayoutConstraint!
     private let style: FileUploadListStyle
-    private let kMaxUnscrollableViews = 3
+    private var maxUnscrollableViews: Int {
+        if UIApplication.shared.preferredContentSizeCategory <= .accessibilityMedium {
```
Comment (igorkravchenko) [inline GliaWidgets/View/Chat/Entry/Upload/FileUploadListView.swift:15]: Let's create managed dependency for UIApplication and inject it here via Environment. This dependency will be within UIKitBased namespace.

File: GliaWidgets/View/Chat/Entry/ChatMessageEntryView.swift [hunk 3]
```diff
@@ -85,6 +88,13 @@ class ChatMessageEntryView: UIView {
         updateTextViewHeight()
     }
 
+    func updateLayout() {
+        environment.gcd.mainQueue.asyncAfter(.now() + 0.1) {
```
Comment (igorkravchenko) [inline GliaWidgets/View/Chat/Entry/ChatMessageEntryView.swift:92]: Let's add comment explaining why this workaround is needed here.

**Review by igorkravchenko — COMMENTED** [review]
Looks good. Left a few comments.

File: GliaWidgets/View/Chat/Entry/Upload/FileUploadListView.swift (hunk 2 above)
Comment (EgorovEI) [reply GliaWidgets/View/Chat/Entry/Upload/FileUploadListView.swift:15]: I've created separate ticket [MOB-1345](https://glia.atlassian.net/browse/MOB-1345) for that fix. I'll fix it when @kuldar-daniel will merge his [PR](https://github.com/salemove/ios-sdk-widgets/pull/305)

File: GliaWidgets/View/Chat/Entry/ChatMessageEntryView.swift (hunk 3 above)
Comment (EgorovEI) [reply GliaWidgets/View/Chat/Entry/ChatMessageEntryView.swift:93]: a76e9a4df03bc57953108d86ec9da7aa9bda9ce1

File: GliaWidgets/Component/PoweredBy/PoweredBy.swift [hunk 4]
```diff
@@ -32,8 +32,14 @@ class PoweredBy: UIView {
 
         logoImageView.image = Asset.gliaLogo.image
 
+        let size = logoImageView.systemLayoutSizeFitting(UIView.layoutFittingCompressedSize)
+        logoImageView.heightAnchor.constraint(
+            equalTo: logoImageView.widthAnchor,
+            multiplier: size.height / size.width
```
[the remark this answers is not in the document]
Comment (EgorovEI) [reply GliaWidgets/Component/PoweredBy/PoweredBy.swift:38]: multiplier is the value by which the value of `equalTo` will be multiplied. We can switch anchors and multiplier value to opposite (height / width -> width / height). Result will be the same.

File: GliaWidgets/GCD.Interface.swift [hunk 5]
```diff
@@ -6,5 +6,6 @@ struct GCD {
 extension GCD {
     struct DispatchQueue {
         var async: (@escaping () -> Void) -> Void
+        var asyncAfter: (DispatchTime, @escaping () -> Void) -> Void
```
[the remark this answers is not in the document]
Comment (EgorovEI) [reply GliaWidgets/GCD.Interface.swift]: 9105ed7ddf5f4235ace9be9aad829582d255965a

File: GliaWidgets/Component/PoweredBy/PoweredBy.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (igorkravchenko) [reply GliaWidgets/Component/PoweredBy/PoweredBy.swift:38]: Ok, cool. Just wanted to be sure this is handled.

**Pull request merged**